package-res/resources/web/test/karma/unit/pentaho/type/boolean.Spec.js

```diff
@@ -0,0 +1,69 @@
+/*!
+ * Copyright 2010 - 2016 Pentaho Corporation.  All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+define([
+    "pentaho/type/boolean",
+    "pentaho/type/Context",
+    "pentaho/i18n!/pentaho/type/i18n/types"
+], function(booleanFactory, Context, bundle) {
+
+    "use strict";
+
+    /*global describe:true, it:true, expect:true, beforeEach:true*/
+
+    describe("pentaho/type/boolean -", function() {
+        it("is a function", function() {
+            expect(typeof booleanFactory).toBe("function");
+        });
+
+        describe("new boolean()", function() {
+            var PentahoBoolean;
+            beforeEach(function () {
+                PentahoBoolean = booleanFactory(new Context());
+            });
+            it("should return an object", function () {
+                expect(typeof new PentahoBoolean(1)).toBe("object");
+            });
+            it("should return a function", function () {
+                expect(typeof PentahoBoolean).toBe("function");
+            });
+            it("should accept 1 as true", function () {
+                expect(new PentahoBoolean(1)._value).toBe(true);
```

Comment:
Hi, the `v` property is also an internal detail used for de-serialization/configuration purposes only. Please use the `value` property in this context.
`valueOf()` will be tested in the `Simple` tests.